<launch>
    <!-- <include file="$(find vive)/launch/vive_world.launch"/> -->

    <arg name="rival1_active" default="false"/>
    <arg name="rival1_tracker" default="trackerD"/>
    <arg name="rival2_active" default="true"/>
    <arg name="rival2_tracker" default="trackerC"/>
    <arg name="lowpass_active_" default="true"/>
    <arg name="alpha_" default="0.2"/>
    <arg name="del_vel_" default="0.8"/>
    <arg name="tole_" default="0.1"/>
    <arg name="freq_" default="20"/>
    <arg name="unit_" default="1"/>
    <arg name="side" default="b"/>
    <arg name="print_info_" default="false"/>

    <group ns="rival1" if="$(arg rival1_active)">
        <include file="$(find vive)/launch/vive_world.launch">
            <arg name="survive3_active" value="$(arg rival1_active)"/>
            <arg name="survive3_tracker" value="$(arg rival1_tracker)"/>
            <arg name="side_" value="$(arg side)"/>
            <arg name="print_info" value="$(arg print_info_)"/>
        </include>

        <node name="vive_rival" pkg="vive" type="vive_rival" output="screen">
            <param name="tracker" type="string" value="$(arg rival1_tracker)"/>  <!-- need to check -->
            <param name="map" type="string" value="survive3_map_avg"/>  <!-- need to check -->
            <param name="world" type="string" value="survive3_world"/>  <!-- need to check -->
            <param name="topic_name" type="string" value="/rival1/odom/tracker"/>  <!-- need to check -->
            <param name="tracker_vel_topic" type="string" value="tracker_vel"/>
            
            <param name="lowpass_active" type="bool" value="$(arg lowpass_active_)"/>
            <param name="alpha" type="double" value="$(arg alpha_)"/>
            <param name="del_vel" type="double" value="$(arg del_vel_)"/>
            <param name="tole" type="double" value="$(arg tole_)"/>
            <!-- <param name="error_tole" type="double" value="0.1"/>  -->
            <!-- <param name="insurance_mode" type="bool" value="true"/> while trun on, fusion the Lidar data to void tracker failure -->

            <param name="freq" type="int" value="$(arg freq_)"/>
            <param name="unit" type="int" value="$(arg unit_)"/>
        </node>
    </group>

    <group ns="rival2" if="$(arg rival2_active)">
        <include file="$(find vive)/launch/vive_world.launch">
            <arg name="survive4_active" value="$(arg rival2_active)"/>
            <arg name="survive4_tracker" value="$(arg rival2_tracker)"/>
            <arg name="side_" value="$(arg side)"/>
            <arg name="print_info" value="$(arg print_info_)"/>
        </include>

        <node name="vive_rival" pkg="vive" type="vive_rival" output="screen">
            <param name="tracker" type="string" value="$(arg rival2_tracker)"/>  <!-- need to check -->
            <param name="map" type="string" value="survive4_map_avg"/>  <!-- need to check -->
            <param name="world" type="string" value="survive4_world"/>  <!-- need to check -->
            <param name="topic_name" type="string" value="/rival2/odom/tracker"/>  <!-- need to check -->
            <param name="tracker_vel_topic" type="string" value="/rival2/tracker_vel"/>
            
            <param name="lowpass_active" type="bool" value="$(arg lowpass_active_)"/>
            <param name="alpha" type="double" value="$(arg alpha_)"/>
            <param name="del_vel" type="double" value="$(arg del_vel_)"/>
            <param name="tole" type="double" value="$(arg tole_)"/>
            <!-- <param name="error_tole" type="double" value="0.1"/> 
            <param name="inserance_mode" type="bool" value="true"/> -->
            
            <param name="freq" type="int" value="$(arg freq_)"/>
            <param name="unit" type="int" value="$(arg unit_)"/>
        </node>
    </group>

</launch>